### PR TITLE
class library: add Complex:-sqrt

### PR DESCRIPTION
**Pull request by joslloand**
## Purpose and Motivation

Addresses issue #5213 sclang: Complex.sqrt is missing

## Types of changes

- New feature

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

File: HelpSource/Classes/Complex.schelp [hunk 1]
```diff
@@ -91,6 +91,18 @@ code::
 cubed(Complex(2, 1))
 ::
 
+method:: sqrt
+Complex square root
+discussion::
+returns the principal root
+code::
```
Comment (elgiano) [inline HelpSource/Classes/Complex.schelp:98]: Maybe these two code blocks could be merged into a single one?

File: SCClassLibrary/Common/Math/Complex.sc [hunk 2]
```diff
@@ -95,6 +95,20 @@ Complex : Number {
 		^Complex(real * denom, imag.neg * denom)
 	}
 
+	sqrt {
+		var mag, sign;
+
+		mag = this.magnitude;
+		sign = (imag.sign + 1).asBoolean.if({ 1 }, { 1.neg });  // +1 >= 0, -1 < 0
```
Comment (elgiano) [inline SCClassLibrary/Common/Math/Complex.sc:102]: I would rewrite this as:
```
sign = if(imag.isPositive) { 1 } { -1 }
```

**Review by elgiano — CHANGES_REQUESTED** [review]
Thanks! Before a more thorough refactoring of Complex primitives, this looks good to me, apart from some minor things :)

File: SCClassLibrary/Common/Math/Complex.sc (hunk 2 above)
Comment (brianlheim) [reply SCClassLibrary/Common/Math/Complex.sc:102]: i had to look it up -- `isPositive` is more like `isNonNegative`, while `isStrictlyPositive` actually tests `n > 0`.

Comment (elgiano) [reply SCClassLibrary/Common/Math/Complex.sc:102]: too bad we don't have isNonNegative for Numbers :(

**Comment by joslloand** [comment]
[Changes](https://github.com/supercollider/supercollider/pull/5220/commits/a56360f487a4e8674591c3eb81484da5e6f8966a) suggested by @elgiano submitted....

**Review by elgiano — APPROVED** [review]
Looks good to me! Thanks!

**Comment by jamshark70** [comment]
BTW: Only just noticed... "WARNING: Extension in '/home/dlm/.local/share/SuperCollider/downloaded-quarks/JoshMisc/AnaUtils.sc' overwrites Complex:sqrt in main class library."

**Comment by joslloand** [comment]
@jamshark70, which is [here](https://github.com/supercollider-quarks/JoshMisc/blob/master/AnaUtils.sc#L475-L486), and uses the trigonometric method.

@joshpar, might want to remove from [JoshMisc/AnaUtils.sc](https://github.com/supercollider-quarks/JoshMisc/blob/master/AnaUtils.sc) when the time comes?

**Comment by joshpar** [comment]
Yep - agree. It’s a times like this I wish we had a way to see if something existed or not.

**Pull request merged**